core/src/main/scala/org/apache/spark/ui/exec/ExecutorThreadDumpPage.scala

```diff
@@ -49,11 +49,29 @@ private[ui] class ExecutorThreadDumpPage(parent: ExecutorsTab) extends WebUIPage
     val maybeThreadDump = sc.get.getExecutorThreadDump(executorId)
 
     val content = maybeThreadDump.map { threadDump =>
-      val dumpRows = threadDump.map { thread =>
+      val dumpRows = threadDump.sortWith {
+        case (threadTrace1, threadTrace2) => {
+          val v1 = if (threadTrace1.threadName.contains("Executor task launch")) 1 else 0
+          val v2 = if (threadTrace2.threadName.contains("Executor task launch")) 1 else 0
+          if (v1 == v2) {
+            threadTrace1.threadName > threadTrace2.threadName
```

Comment:
According to the ScalaDoc for `sortWith`, the function passed to `sortWith` is

> the comparison function which tests whether its first argument precedes its second argument in the desired ordering.

Here, I think that we want all executor threads to appear before any non-executor thread.  Within each group, I think that we intuitively want alphabetical ordering, but here it looks like we're returning reverse-alphabetical ordering. Therefore, I think that the `>` should actually be a `<` here.